scripts/core/editor3/helpers/handleBeforeInputHighlights.ts

```diff
@@ -15,28 +21,40 @@ function isSelectionAtEndOfBlock(editorState: EditorState): boolean {
 }
 
 /**
- * @ngdoc method
- * @name addCommentsForServer
- * @param {EditorState} editorState
- * @param {SelectionState} collapsedSelection
- * @return {List<CharacterMetadata>}
+ * Returns the characters before and after the cursor.
```

Comment:
Your comment on Github was clearer. This one doesn't explain why empty chars are pushed to the list.